### PR TITLE
feature: routesv HEAD response with X-Count header

### PR DESCRIPTION
**Pull request by szuecs**
Signed-off-by: Sandor Szücs <sandor.szuecs@zalando.de>

### CODE REVIEW

File: routesrv/eskipbytes.go [hunk 1]
```diff
@@ -61,11 +65,23 @@ func (e *eskipBytes) ServeHTTP(w http.ResponseWriter, r *http.Request) {
...
 		w.Header().Add("Etag", etag)
 		w.Header().Add("Content-Type", "text/plain; charset=utf-8")
-		http.ServeContent(w, r, "", lastModified, bytes.NewReader(data))
+		w.Header().Add(routing.RoutesCountName, strconv.Itoa(e.count))
+
+		switch r.Method {
+		case http.MethodHead:
+			w.WriteHeader(http.StatusOK)
+		case http.MethodGet:
+			http.ServeContent(w, r, "", lastModified, bytes.NewReader(data))
```
Comment (AlexanderYastrebov) [inline routesrv/eskipbytes.go:83]: http.ServeContent supports HEAD method so I think we'd only need to add X-Count header

File: routesrv/eskipbytes.go [hunk 2]
```diff
@@ -61,11 +65,23 @@ func (e *eskipBytes) ServeHTTP(w http.ResponseWriter, r *http.Request) {
...
+		w.WriteHeader(http.StatusMethodNotAllowed)
+		return
+	}
+
 	data, etag, lastModified, initialized := e.bytes()
 	if initialized {
 		w.Header().Add("Etag", etag)
 		w.Header().Add("Content-Type", "text/plain; charset=utf-8")
-		http.ServeContent(w, r, "", lastModified, bytes.NewReader(data))
+		w.Header().Add(routing.RoutesCountName, strconv.Itoa(e.count))
```
Comment (AlexanderYastrebov) [inline routesrv/eskipbytes.go:77]: Here `e.count` is read without holding the mutex. Either `e.bytes()` should be extended to return count or we should drop it and lock/unclock mutex manually

Comment (szuecs) [reply routesrv/eskipbytes.go]: done

**Comment by AlexanderYastrebov** [comment]
:+1:

**Comment by szuecs** [comment]
:+1:

**Pull request merged**